@gerritjvv When you see unhelpful error messages like that, it is very useful to run your build with `--info`, which will print out the error messages from the test builds.  In this case, I get: 
```
    11:17:18.399 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
    11:17:18.401 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] FAILURE: Build failed with an exception.
    11:17:18.401 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
    11:17:18.401 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * What went wrong:
    11:17:18.402 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Execution failed for task ':ensime'.
    11:17:18.402 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > Could not resolve all dependencies for configuration ':ensime'.
    11:17:18.402 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]    > Cannot resolve external dependency org.ensime:server_2.11:1.0.1 because no repositories are defined.
    11:17:18.402 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]      Required by:
    11:17:18.402 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]          project :
```

--- a/src/main/groovy/net/coacoas/gradle/plugins/Ensime.groovy
+++ b/src/main/groovy/net/coacoas/gradle/plugins/Ensime.groovy
@@ -70,6 +70,12 @@ class EnsimeModel {
   File javaHome
   String cacheDir = ""
 
+  //The ensime server jar and dependencies are downloaded to this location
+  //then added to the ensime-server-jars property in the .ensime file
+  String serverJarsDir = "build/ensime"
+
+  String serverVersion = "1.0.1"
+  
   List<String> javaFlags = []
   List<String> referenceSourceRoots = []
   List<String> compilerArgs = []
@@ -87,6 +93,8 @@ class EnsimeModel {
             ", scalaVersion=" + scalaVersion +
             ", javaHome=" + javaHome +
             ", cacheDir=" + cacheDir +
+            ", serverJarsDir=" + serverJarsDir +
+            ", serverVersion=" + serverVersion +
             ", javaFlags=" + javaFlags +
             ", referenceSourceRoots=" + referenceSourceRoots +
             ", compilerArgs=" + compilerArgs +

--- a/src/main/groovy/net/coacoas/gradle/plugins/EnsimeTask.groovy
+++ b/src/main/groovy/net/coacoas/gradle/plugins/EnsimeTask.groovy
@@ -7,6 +7,7 @@ import org.gradle.api.artifacts.DependencySet
 import org.gradle.api.artifacts.UnknownConfigurationException
 import org.gradle.api.tasks.TaskAction
 
+
 /*
  * Implementation of the 'ensime' task.
  */
@@ -22,9 +23,54 @@ class EnsimeTask extends DefaultTask {
     File outputFile = ensimeFile(project.extensions.ensime.targetFile)
 
     project.extensions.ensime.with { EnsimeModel model ->
+
+      project.repositories.mavenLocal()
+      project.repositories.mavenCentral()
+
+      //if snapshot is used, include the snapshot repository
+      //this avoids having to manually edit the build script
+      if(model.serverVersion.contains("SNAPSHOT")){
+	project.repositories.maven {
+		name 'Maven sonatype snapshots'
+		url 'https://oss.sonatype.org/content/repositories/snapshots'	
+	}
+      }
+      else{
+	project.repositories.maven {
+		name 'Maven sonatype snapshots'
+		url 'https://oss.sonatype.org/content/repositories/public'
+	}
+      }
+      
+      project.configurations.create('ensime')
+      project.configurations.ensime.transitive = true
+
+      def scalaVersion = (model.scalaVersion ?: findScalaVersion(project))
+      def m = scalaVersion =~ /\d+\.\d+/
+      def serverScalaVersion = m.count ? m[0] : null
+      
+      project.dependencies.add('ensime', "org.ensime:server_${serverScalaVersion}:${model.serverVersion}")
+
       project.logger.with { logger ->
         logger.info("Ensime Model: ${model}")
 
+        logger.debug("Ensime dependencies: ${project.configurations.ensime.files}")
+      
+        logger.debug("Tools: ${this.findJavaHome(model)}/lib/tools.jar")
+	
+        //copy all ensime server dependencies
+        //see ~/.gradle/init.gradle AddEnsimePlugin
+        project.copy {
+          from(project.configurations.ensime.files)
+          into(model.serverJarsDir)
+        }
+        //copy the java tools.jar
+        project.copy {
+	  from("${this.findJavaHome(model)}/lib/tools.jar")
+          into(model.serverJarsDir)
+        }
+      
+    
         // root-dir ...
         assert !project.rootDir.absolutePath.empty : "root-dir must be not empty"
         properties.put("root-dir", project.rootDir.absolutePath)
@@ -42,6 +88,10 @@ class EnsimeTask extends DefaultTask {
         properties.put("cache-dir", ensimeCacheDir)
         logger.debug("EnsimeTask: Writing cache-dir: ${ensimeCacheDir}")
 
+	def serverJars = new File("build/ensime").listFiles().collect { it.absolutePath }
+	properties.put("ensime-server-jars", serverJars)
+	logger.debug("ensimeTask: Writing ensime-server-jars: ${serverJars}")
+	
         // (project) name ...
         assert !project.name.empty, "project.name must be not empty"
         properties.put("name", project.name)
@@ -62,8 +112,8 @@ class EnsimeTask extends DefaultTask {
         logger.debug("EnsimeTask: Writing reference-source-roots: ${model.referenceSourceRoots}")
 
         // scala-version ...
-        properties.put("scala-version", this.findScalaVersion(project))
-        logger.debug("EnsimeTask: Writing scala-version: ${model.scalaVersion}")
+        properties.put("scala-version", scalaVersion)
+        logger.debug("EnsimeTask: Writing scala-version: ${scalaVersion}")
 
         // compiler-args ...
         properties.put("compiler-args", model.compilerArgs)

--- a/src/main/groovy/net/coacoas/gradle/plugins/EnsimeTask.groovy
+++ b/src/main/groovy/net/coacoas/gradle/plugins/EnsimeTask.groovy
@@ -35,7 +35,7 @@ class EnsimeTask extends DefaultTask {
 		url 'https://oss.sonatype.org/content/repositories/snapshots'	
 	}
       }
-      else{
+      else {
 	project.repositories.maven {
 		name 'Maven sonatype snapshots'
 		url 'https://oss.sonatype.org/content/repositories/public'

--- a/src/test/groovy/net/coacoas/gradle/plugins/SimpleProjectTest.groovy
+++ b/src/test/groovy/net/coacoas/gradle/plugins/SimpleProjectTest.groovy
@@ -47,7 +47,9 @@ public class SimpleProjectTest extends Specification implements ProjectSpecifica
         configuration.contains(javaVersion)
         configuration.contains(":java-flags ()")
 
-        !configuration.contains(':runtime-deps')
+	configuration.find { it.matches(":ensime-server-jars.*") }
+
+	!configuration.contains(':runtime-deps')
 
         configuration.find { it =~ /^:compile\-deps\s+\(".*scala\-library\-2\.10\.4\.jar"\)/ }
 
@@ -73,11 +75,17 @@ public class SimpleProjectTest extends Specification implements ProjectSpecifica
         result.output.contains("Using Scala version 2.11.7")
         File ensime = new File(testProjectDir.root, '.ensime')
         ensime.exists()
-        String configuration = ensime.readLines()
+
+        List<String> configuration = ensime.readLines()
         configuration.contains(":scala-version \"2.11.7\"")
-        configuration.contains(":java-home \"${javaHome()}\"")
+
+        String javaVersion = ":java-home \"${javaHome()}\""
+        configuration.contains(javaVersion)
         configuration.contains(":java-flags ()")
-        !configuration.contains(':runtime-deps')
+
+	configuration.find { it.contains(':ensime-server-jars') }
+
+	!configuration.contains(':runtime-deps')
 
         where:
         gradleVersion << supportedVersions

